src/App/Fossa/Ficus/Analyze.hs

```diff
@@ -53,6 +68,13 @@ newtype CustomLicensePath = CustomLicensePath {unCustomLicensePath :: Text}
 newtype CustomLicenseTitle = CustomLicenseTitle {unCustomLicenseTitle :: Text}
   deriving (Eq, Ord, Show, Hashable)
 
+-- Helper function to log with timestamp
+logDebugWithTime :: (Has Logger sig m, Has (Lift IO) sig m) => Text -> m ()
+logDebugWithTime msg = do
+  now <- sendIO getCurrentTime
+  let timestamp = formatTime defaultTimeLocale "%H:%M:%S.%3q" now
+  logDebug $ "[" <> pretty timestamp <> "] " <> pretty msg
```

Comment:
[nit] Optional, but may be helpful:
```suggestion
logDebugWithTime :: (Has Logger sig m, Has (Lift IO) sig m) => Text -> m a -> m a
logDebugWithTime msg action = do
  let writeTime = do now <- sendIO getCurrentTime
                                 let timestamp = formatTime defaultTimeLocale "%H:%M:%S.%3q" now
                                 logDebug $ "[" <> pretty timestamp <> "] " <> pretty msg
  writeTime
  action
  writeTime
```

The idea is it will write a start/end. You could even make it calculate a duration as a summary.